comp/core/tagger/taggerimpl/tagger_test.go

```diff
@@ -35,52 +38,117 @@ func (f *fakeCIDProvider) ContainerIDForPodUIDAndContName(podUID, contName strin
 	return f.entries[id], nil
 }
 
+// Sets up the fake metrics provider and returns a function to reset the original metrics provider
+func setupFakeMetricsProvider(mockMetricsProvider metrics.Provider) func() {
+	originalMetricsProvider := metrics.GetProvider
+	metrics.GetProvider = func(_ optional.Option[workloadmeta.Component]) metrics.Provider {
+		return mockMetricsProvider
+	}
+	return func() { metrics.GetProvider = originalMetricsProvider }
+}
+
 func TestEnrichTags(t *testing.T) {
 	// Create fake tagger
 	fakeTagger := fxutil.Test[tagger.Mock](t, MockModule())
+	containerName, initContainerName, containerID, initContainerID, podUID := "container-name", "init-container-name", "container-id", "init-container-id", "pod-uid"
 
 	// Fill fake tagger with entities
-	fakeTagger.SetTags(types.NewEntityID(types.KubernetesPodUID, "pod"), "host", []string{"pod-low"}, []string{"pod-orch"}, []string{"pod-high"}, []string{"pod-std"})
-	fakeTagger.SetTags(types.NewEntityID(types.ContainerID, "container"), "host", []string{"container-low"}, []string{"container-orch"}, []string{"container-high"}, []string{"container-std"})
+	fakeTagger.SetTags(types.NewEntityID(types.KubernetesPodUID, podUID), "host", []string{"pod-low"}, []string{"pod-orch"}, []string{"pod-high"}, []string{"pod-std"})
+	fakeTagger.SetTags(types.NewEntityID(types.ContainerID, containerID), "host", []string{"container-low"}, []string{"container-orch"}, []string{"container-high"}, []string{"container-std"})
+	fakeTagger.SetTags(types.NewEntityID(types.ContainerID, initContainerID), "host", []string{"init-container-low"}, []string{"init-container-orch"}, []string{"init-container-high"}, []string{"init-container-std"})
 
+	// Local data tests
 	for _, tt := range []struct {
 		name         string
 		originInfo   taggertypes.OriginInfo
-		cidProvider  *fakeCIDProvider
```

Comment:
Imo you should keep the fake tagger part of the unittest struct.
It helps to start each test with a clean instance of the tagger mock. So you are sure a previous test doesn't leak some data in the tagger. 

maybe the `fxutil.Test()` function is able to automatically cleanup the mock between each test. but in that case the initialisation of the tagger should be part of the sub test (`tt`) execution. 

With the current tests case, the change is not needed because the tests are using the tagger as "static and read only tagger", but for some other test use cases, it requires to instance the mock in each test, even more when test can be executed in //.

Reply:
I can add the *fakeCIDProvider back but I don't understand your comment fully. You might've conflicted the fakeCIDProvider with the fakeTagger?

I removed the fakeCIDProvider because it was doing nothing in that test case. From what I understand, the fakeCIDProvider was a struct created to specifically just test `generateContainerIDFromExternalData` because it takes `provider.ContainerIDForPodUIDAndContNameRetriever` as a parameter which matches the struct `fakeCIDProvider`.

In this test case we are testing `EnrichTags` which is "static and read only". This is why I believe whoever created `TestEnrichTags` originally didn't need to define the `fakeTagger` at the subtest level.